optuna/samplers/_tpe/sampler.py

```diff
@@ -233,6 +310,38 @@ def _split_observation_pairs(
         above = np.asarray([v for v in above if v is not None], dtype=float)
         return below, above
 
+    def _split_multivariate_observation_pairs(
+        self,
+        config_vals: Dict[str, List[Optional[float]]],
+        loss_vals: List[Tuple[float, float]],
+    ) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+
+        config_vals = {k: np.asarray(v, dtype=float) for k, v in config_vals.items()}
+        loss_vals = np.asarray(loss_vals, dtype=[("step", float), ("score", float)])
+
+        # TODO(kstoneriv3): Change of the order of 1. exclusion of `None` and 2. splitting might
+        # be needed. Independent sampler in `TPESampler` first splits the observations and
+        # then exclude `None`.
+
+        # We exclude param_vals with `None`.
+        config_vals_matrix = np.array([v for v in config_vals.values()])
+        index_none = np.any(np.equal(config_vals_matrix, None), axis=0)
+        config_vals = {k: v[~index_none] for k, v in config_vals.items()}
+        loss_vals = loss_vals[~index_none]
```

Comment:
Here, we exclude the trials which have a `None` among any parameters. I think that this means that we cannot sample based on the case studies. As a default, the search space is the `IntersectionSearchSpace`, so it is OK in this PR not to think the case-based sampling, but I think it is not a good idea to prohibit the case-based sampling in the revel of the sampler logic. (It should be in the revel of the search space logic.)  Improving this strategy can be an important future work. How about mentioning it as comments?
```suggestion
        # We exclude trials with `None` values among any parameters. 
        # This means that we cannot sample based on case studies.
        # Improving this strategy is a future work.
        config_vals_matrix = np.array([v for v in config_vals.values()])
        index_none = np.any(np.equal(config_vals_matrix, None), axis=0)
        config_vals = {k: v[~index_none] for k, v in config_vals.items()}
        loss_vals = loss_vals[~index_none]
```

Reply:
Another idea of comments:
"We exclude trials with `None` values among any parameters. This means that we transfer the search space into the instance of `InterSectionSpace`."

Reply:
The first sentence sounds good. What do you think about using the following as the second sentence?
"This is equivalent to converting the search space into the instance of `InterSectionSpace`."